Add example of tune calculation (postprocessing) (#516)
BLAST-ImpactX/impactx: examples/fodo_tune/README.rst

.. _examples-fodo-tune:

Tune Calculation in a Periodic FODO Channel

> ============================================

Reviewer: ```suggestion
Tune Calculation in a Periodic FODO Channel
===========================================
```

> This is identical to the FODO example, except that tracking for 100 periods is used to extract the horizontal tune.

Reviewer: ```suggestion
This is identical to the :ref:`FODO example <examples-fodo>`, except that tracking for 100 periods is used to extract the horizontal tune.
```

Stable FODO cell with a zero-current phase advance of 67.8 degrees, corresponding to Qx = Qy = 0.1883.

The matched Twiss parameters at entry are:

* :math:`\beta_\mathrm{x} = 2.82161941` m
* :math:`\alpha_\mathrm{x} = -1.59050035`
* :math:`\beta_\mathrm{y} = 2.82161941` m
* :math:`\alpha_\mathrm{y} = 1.59050035`

We use a 2 GeV electron beam with initial unnormalized rms emittance of 2 nm.

The horizontal tune of a single particle is obtained from period-by-period tracking data using several algorithms.

In this test, the computed horizontal tune must agree with the nominal value to within acceptable tolerance.

> This example requires installation of PyNAFF:  https://pypi.org/project/PyNAFF/

Reviewer: ```suggestion
This example requires installation of `PyNAFF <https://github.com/nkarast/PyNAFF>`__.
```

> This example requires installation of PyNAFF:  https://pypi.org/project/PyNAFF/

Reviewer: I checked the requirements on PyNAFF: https://github.com/nkarast/PyNAFF/blob/master/requirements.txt

They are a subset of ours, so we could easily make it a mandatory dependency for Python builds in our `requirements.txt` :)

Author: That's fantastic.  I have some ideas about a few additional testing/details to add to the script, so it isn't finalized (but it works for me with PyNAFF installed).

Also, it's still worth exploring independently some methods to calculate the tunes in-situ.
